### PR TITLE
114 support keypair generation for curve448 montgomerry and edwards forms

### PR DESCRIPTION
**Pull request by JosePisco**
Support for Curve448 both Montgomerry and Edwards forms:

- keypair generation
- kmip keypair conversion
- cryptographic algorithms selection in kms

### CODE REVIEW

File: crate/kmip/src/openssl/private_key.rs [hunk 1]
```diff
@@ -154,6 +154,12 @@ pub fn kmip_private_key_to_openssl(private_key: &Object) -> Result<PKey<Private>
                     pad_be_bytes(&mut privkey_vec, 32);
                     PKey::private_key_from_raw_bytes(&privkey_vec, Id::ED25519)?
                 }
+                RecommendedCurve::CURVEED448 => {
+                    let mut privkey_vec = d.to_bytes_be();
+                    // 57 is privkey size on Ed448.
+                    pad_be_bytes(&mut privkey_vec, 57);
```
Comment (Manuthor) [inline crate/kmip/src/openssl/private_key.rs:160]: we should use constant from kms_utils for all privkey sizes. Quite a few hardcoded values could be replaced.

File: crate/kmip/src/openssl/private_key.rs [hunk 2]
```diff
@@ -755,6 +761,216 @@ mod tests {
         assert_eq!(private_key_.bits(), 253);
     }
 
+    #[test]
+    fn test_ec_ed25519_private_key() {
```
Comment (Manuthor) [inline crate/kmip/src/openssl/private_key.rs:765]: those 4 tests are very similar, should we write a function?
- `test_ec_x25519_private_key`
- `test_ec_ed25519_private_key`
- `test_ec_x448_private_key`
- `test_ec_ed448_private_key`

File: crate/kmip/src/openssl/private_key.rs [hunk 3]
```diff
@@ -755,6 +761,216 @@ mod tests {
...
+        let private_key_ = PKey::private_key_from_pkcs8(&key_value).unwrap();
+        assert_eq!(private_key_.id(), Id::X448);
+        assert_eq!(private_key_.bits(), 448);
+        assert_eq!(private_key_.private_key_to_pkcs8().unwrap(), key_value);
+        let private_key_ = kmip_private_key_to_openssl(&object_).unwrap();
+        assert_eq!(private_key_.id(), Id::X448);
+        assert_eq!(private_key_.bits(), 448);
+        assert_eq!(private_key_.private_key_to_pkcs8().unwrap(), key_value);
+
+        // SEC1 is not available for X25519
```
Comment (Manuthor) [inline crate/kmip/src/openssl/private_key.rs:865]: update comment

**Review by Manuthor — COMMENTED** [review]
great job!

File: crate/kmip/src/openssl/private_key.rs (hunk 1 above)
Comment (JosePisco) [reply crate/kmip/src/openssl/private_key.rs:160]: I strongly share this opininon. However, utils depends on kmip which makes it impossible to depend on utils.
I'm open to ideas :)

File: crate/kmip/src/openssl/private_key.rs (hunk 3 above)
Comment (JosePisco) [reply crate/kmip/src/openssl/private_key.rs]: c235c9b60ea8af0ada1dc9397fbe6495be25cf5f

File: crate/kmip/src/openssl/private_key.rs (hunk 2 above)
Comment (JosePisco) [reply crate/kmip/src/openssl/private_key.rs]: I propose commit c235c9b60ea8af0ada1dc9397fbe6495be25cf5f for a refacto of these tests. If it is accepted, I will use this pattern to refacto public key tests as well.

Comment (JosePisco) [reply crate/kmip/src/openssl/private_key.rs]: commit 7f42f9ee61e3789c16fe7b55a33c5e34be132c94 should improve refacto and do it for public keys.

**Pull request merged**